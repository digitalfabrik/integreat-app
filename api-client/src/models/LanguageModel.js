// @flow

import type { LanguageDirectionType } from '../types'

class LanguageModel {
  _code: string
  _name: string
  _direction: ?LanguageDirectionType

  constructor (code: string, name: string, direction?: LanguageDirectionType) {
    this._code = code
    this._name = name
    this._direction = direction
  }

  get code (): string {
    return this._code
  }

  get name (): string {
    return this._name
  }

  get direction (): ?LanguageDirectionType {
    return this._direction
  }
}

export default LanguageModel
