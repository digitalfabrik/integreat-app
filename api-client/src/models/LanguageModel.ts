import type { LanguageDirectionType } from '../types'

class LanguageModel {
  _code: string
  _name: string
  _direction: LanguageDirectionType | null | undefined

  constructor(code: string, name: string, direction?: LanguageDirectionType) {
    this._code = code
    this._name = name
    this._direction = direction
  }

  get code(): string {
    return this._code
  }

  get name(): string {
    return this._name
  }

  get direction(): LanguageDirectionType | null | undefined {
    return this._direction
  }
}

export default LanguageModel