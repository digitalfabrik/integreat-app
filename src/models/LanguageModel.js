// @flow

class LanguageModel {
  _code: string
  _name: string

  constructor (code: string, name: string) {
    this._code = code
    this._name = name
  }

  get code (): string {
    return this._code
  }

  get name (): string {
    return this._name
  }
}

export default LanguageModel
