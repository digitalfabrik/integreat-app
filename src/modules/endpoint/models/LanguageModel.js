class LanguageModel {
  constructor (code, name) {
    this._code = code
    this._name = name
  }

  get code () {
    return this._code
  }

  get name () {
    return this._name
  }
}

export default LanguageModel
