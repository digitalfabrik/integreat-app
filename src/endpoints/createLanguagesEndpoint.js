// @flow

import LanguageModel from '../models/LanguageModel'
import EndpointBuilder from '../EndpointBuilder'
import ParamMissingError from '../errors/ParamMissingError'
import type { JsonLanguageType } from '../types'
import Endpoint from '../Endpoint'

export const LANGUAGES_ENDPOINT_NAME = 'languages'

type ParamsType = { city: ?string }

export default (baseUrl: string): Endpoint<ParamsType, Array<LanguageModel>> =>
  new EndpointBuilder(LANGUAGES_ENDPOINT_NAME)
    .withParamsToUrlMapper(params => {
      if (!params.city) {
        throw new ParamMissingError(LANGUAGES_ENDPOINT_NAME, 'city')
      }
      return `${baseUrl}/${params.city}/de/wp-json/extensions/v3/languages`
    })
    .withMapper((json: Array<JsonLanguageType>) => json
      .map((language: JsonLanguageType) => new LanguageModel(
        language.code,
        language.native_name,
        language.dir
      ))
      .sort((lang1, lang2) => lang1.code.localeCompare(lang2.code))
    )
    .build()
