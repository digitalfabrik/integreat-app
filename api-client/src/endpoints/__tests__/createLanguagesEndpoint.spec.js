// @flow

import createLanguagesEndpoint from '../createLanguagesEndpoint'
import LanguageModel from '../../models/LanguageModel'

describe('languages', () => {
  const baseUrl = 'https://integreat-api-url.de'
  const languages = createLanguagesEndpoint(baseUrl)

  const languagesJson = [
    {
      code: 'en',
      native_name: 'English',
      dir: 'ltr'
    },
    {
      code: 'de',
      native_name: 'Deutsch',
      dir: 'ltr'
    },
    {
      code: 'ar',
      native_name: '\u0627\u0644\u0639\u0631\u0628\u064a\u0629',
      dir: 'rtl'
    }
  ]

  const params = {city: 'augsburg'}

  it('should map router to url', () => {
    expect(languages.mapParamsToUrl(params)).toEqual(
      'https://integreat-api-url.de/augsburg/de/wp-json/extensions/v3/languages'
    )
  })

  it('should throw if the city to map the url are missing', () => {
    expect(() => languages.mapParamsToUrl({city: undefined, language: 'de'})).toThrowErrorMatchingSnapshot()
  })

  it('should map fetched data to models', () => {
    const languageModels = languages.mapResponse(languagesJson, params)
    expect(languageModels).toEqual([
      new LanguageModel('ar', '\u0627\u0644\u0639\u0631\u0628\u064a\u0629', 'rtl'),
      new LanguageModel('de', 'Deutsch', 'ltr'),
      new LanguageModel('en', 'English', 'ltr')
    ])
  })
})
