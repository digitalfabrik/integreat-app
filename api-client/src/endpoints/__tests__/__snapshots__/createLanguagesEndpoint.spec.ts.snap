// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languages should throw if the city to map the url are missing 1`] = `"ParamMissingError: Failed to load the languages endpoint because the city is missing"`;
